templates.go

```diff
@@ -66,42 +66,44 @@ import (
 	"google.golang.org/grpc/test/bufconn"
 )
 {{range $lis := .Listeners}}
-var (
-	// {{$lis}} is a global in-memory buffer that listeners that is
+type {{$lis | UpperCase }} struct {
+	// listener is a global in-memory buffer listener that is
 	// referenced by the generated mobile APIs, such that all client calls
 	// will be going through it.
-	{{$lis}} = bufconn.Listen(100)
+	listener *bufconn.Listener
 
-	// {{$lis}}DialOptions holds extra grpc options we'll apply
+	// dialOptions holds extra grpc options we'll apply
 	// every time we dial the grpc server, such as TLS certificates.
-	{{$lis}}DialOptions []grpc.DialOption
+	dialOptions []grpc.DialOption
 
-	// {{$lis}}OptionsMtx is a mutex used to grant exclusive acces
+	// optionsMtx is a mutex used to grant exclusive access
 	// to the options slice.
-	{{$lis}}OptionsMtx sync.Mutex
-)
+	optionsMtx sync.Mutex
+}
+
+var {{$lis}} {{$lis | UpperCase}}
```

Comment:
This will still be global. 

I think we rather should make the `NewListeners` method return a new struct that holds the listeners we need, and then we can pass these listeners to the methods that use them.

Reply:
@halseth That definitely makes sense.

Is it possible though? Inside [`bindings.go`](https://github.com/lightningnetwork/lnd/blob/b4bf4b2906c066ec0d0b8d7183c720b1d7d19220/mobile/bindings.go#L71-L80) they could be passed to the lnd conf so that the listeners get properly set up in lnd's gPRC side of things.

However we also have the actual generated gRPC functions that are directly called from the host application.
These functions are currently calling `get<listener>Conn()`:
https://github.com/lightninglabs/falafel/blob/1bfcffa7997475acca5c0ad10100fb6001dd7d55/templates.go#L39-L40

I don't see how we could prevent some kind of global scope as we cannot pass the listeners back to the host, thus we must set a state inside falafel's gRPC files.

Reply:
Ah, I see. The generated gRPC functions are parsed by gomobile and converted to Java or Objective-C compatible bindings, and they are currently "statitc", in that they reference the running LND through the global variables.

To achieve what I initially thought, we would need these methods to reference a local LND instance, so we could all them like this:
```go
func (l *Lnd) GetInfo(...) (...) {

}
```
 I haven't tried this so not sure if it even is supported by gomobile passing the LND reference across the FFI boundary like that. If not, then I think you are right that it would be easier just sticking to globals.

Reply:
Yes precisely.

I'm not sure that passing the LND reference across the FFI boundary is even preferable, I don't have any particular argument against it but it sounds like an unsound idea.

Perhaps this could be addressed a later point as well.
My suggestion is to go with the solution we have currently. It's still an improvement albeit not quite perfect.

Reply:
Yep, agreed. It is a step in the right direction 👍